Fixed crash if player restarts mission in SP.

nukeulater opened the pull request:
Also fixed the onGameEngineChange due to bad merge.
Some cleanup.
Added missing Initialize for H2 final gametype.

nukeulater committed:
Fixed crash if player restarts mission in SP.

Also fixed the onGameEngineChange due to bad merge.
Some cleanup.

shayaantx commented on xlive/H2Tweaks.cpp:
```diff
@@ -414,3 +414,13 @@ void H2Tweaks::FixRanksIcons() {
 
+void applyCrashFix() {
+
+	//Crashfix
```
if you're gonna comment this out, why keep it?

nukeulater replied:
will fix

shayaantx commented on xlive/H2MOD.cpp:
```diff
@@ -1273,7 +1229,7 @@ void H2MOD::Initialize()
 		PatchGameDetailsCheck();
 		//H2Tweaks::PatchPingMeterCheck();
-		*(bool*)((char*)h2mod->GetBase() + 0x422450) = 1; //allows for all live menus to be accessed
+		*(bool*)(h2mod->GetBase() + 0x422450) = 1; //allows for all live menus to be accessed
```
why you change this?

nukeulater replied:
no need of type-casting it to char* then to dereference it as bool, also i'm the one who added that a while ago.

shayaantx commented:
who tested this?

NukeULater commented:
me and general.

nukeulater committed:
Disabled the "Dedicated command" message.

nukeulater committed:
Small change, cleanup

nukeulater committed:
Merge branch 'development' into crash_fix_1

nukeulater committed:
Merge remote-tracking branch 'PermaNulled/development' into crash_fix_1

nukeulater committed:
Added get_engine_type to use in onGameEngineChange.

GetBase inline.
small clean-up.

nukeulater committed:
Formatting change.

nukeulater committed:
Changes after review

Also fixed wrong variable type -__-

shayaantx commented on xlive/H2MOD.cpp:
```diff
@@ -229,7 +231,7 @@ void H2MOD::logToDedicatedServerConsole(wchar_t* message) {
 
 int __cdecl dediCommandHook(int a1, int a2, int a3) {
-	h2mod->logToDedicatedServerConsole(L"Dedicated command\n");
+	//h2mod->logToDedicatedServerConsole(L"Dedicated command\n");
```
why?

nukeulater replied:
Would be nice to log that only if a custom command has been used.

nukeulater committed:
Applied the changes requested by Num

Also display message in dedicated server console only if the custom command doesn't exist.

nukeulater committed:
Deleted some unused variables.

Deleted unused variables that were providing functionality to the old, now deprecated udp network thread. Clean-up.

shayaantx commented on xlive/H2ConsoleCommands.cpp:
```diff
@@ -465,6 +465,7 @@ void ConsoleCommands::handle_command(std::string command) {
 		}
 		else {
 			output(L"Unknown command.");
+			h2mod->logToDedicatedServerConsole(L"No such custom command.");
```
i looked at code, output will output to console, u can just remove this

nukeulater replied:
alright, thanks

nukeulater committed:
Applied changes requested by Yoshi

Also deleted some unused function definitions.

nukeulater committed:
Moved GameEngine.h to the appropriate folder
